### PR TITLE
Added SMTP email support

### PR DESCRIPTION
**Pull request by shafiq-muhammad**
Adding the support for SMTP email support

### CODE REVIEW

**Comment by curita** [comment]
Hi @shafiq-muhammad!

Some considerations:
1. The CI build for Python 3.9, those errors should be fixed
2. About adding tests, I found there's a flag `debug` that can be sent to `MailSender()` so no actual messages are sent ([reference](https://github.com/scrapy/scrapy/blob/ded28f7bb908090952c066a782c8518efaf2dd79/scrapy/mail.py#L35)). This is used in Scrapy's tests with a callback to catch the message ([reference](https://github.com/scrapy/scrapy/blob/master/tests/test_mail.py)). Maybe that's something we could replicate to add tests for the new `SendSmtpEmail()`?

    We could add a `debug` argument to `SendSmtpEmail.__init__()` or `SendSmtpEmail.send_message()`, and then pass that flag to `MailSender()`. We don't need to test `MailSender()` much as that's already handled by Scrapy, we can focus here on making sure we are instantiating it correctly and calling its send method as expected, and that the message gets "delivered", by using that callback or capturing the debug log messages.

Let me know if that helps and ping me for anything else I can assist you with!

File: spidermon/contrib/actions/email/smtp.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                "smtp_password": crawler.settings.get("SPIDERMON_SMTP_PASSWORD"),
+                "smtp_ssl": crawler.settings.get("SPIDERMON_SMTP_SSL")
+            }
+        )
+        return kwargs
+
+    def send_message(self, message, **kwargs):
+        if not self.sender:
+            raise NotConfigured("You must provide the sender.")
+        if not self.to:
```
Comment (rennerocha) [inline spidermon/contrib/actions/email/smtp.py:52]: **issue** Recipient check is [already validated](https://github.com/scrapinghub/spidermon/blob/master/spidermon/contrib/actions/email/__init__.py#L57) inside `SendEmail` definition.

File: spidermon/contrib/actions/email/ses.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def _get_recipients(self):
                 recipients.extend(recipient_subset)
         return recipients
 
-    def send_message(self, message):
+    def send_message(self, message, **kwargs):
```
Comment (rennerocha) [inline spidermon/contrib/actions/email/ses.py:57]: **question** Why are you changing this class that is not related to the issue you are implementing?

**Review by rennerocha — CHANGES_REQUESTED** [review]
Added some comments and suggestions into the PR.

Regarding the tests, I agree with @curita . We don't need to test the internals of `MailSender` that is already handled by Scrapy. You need to test if `MailSender.send()` is called using the right values according to the settings provided (search for assert_called_with in pytest docs).

Another test that is missing is to ensure that the `NotConfigured` is raised for each of the settings are not provided ([an example can be found here](https://github.com/scrapinghub/spidermon/blob/master/tests/contrib/actions/telegram/test_telegram_action.py#L50).

File: spidermon/contrib/actions/email/ses.py (hunk 2 above)
Comment (shafiq-muhammad) [reply spidermon/contrib/actions/email/ses.py:57]: I needed to add callback and setting some argument such as debug in MailSender while doing testing. So I've to change the base send_message and to make sure that this class doesn't break with that change. That's why I've changed this method.

File: spidermon/contrib/actions/email/smtp.py (hunk 1 above)
Comment (shafiq-muhammad) [reply spidermon/contrib/actions/email/smtp.py:52]: Yes, That's correct. I'll remove check on that.

**Comment by rennerocha** [comment]
This PR solves #337

**Comment by rennerocha** [comment]
Hi @curita ! Yes, I am trying to finish this PR so we can have a new release of Spidermon (I need some new features!).
I added a few more settings options (that are available in Scrapy MailSender class) and added the documentation. It is only missing an update on the tests (current tests are testing MailSender, not the action). I should be finishing this by this week.

**Comment by curita** [comment]
Thank you for the update and your work here again 🙇‍♀️ Sounds good then, if you need any help let us know!
